src/software/ai/primitive/primitive.h

```diff
@@ -46,3 +46,13 @@ class Primitive
 
     virtual ~Primitive() = default;
 };
+
+/**
+ * We use this typedef to pass lists of Primitives around
+ *
+ * Since `std::unique_ptr<Primitive>` does not have a copy constructor, we
+ * need to pass a `shared_ptr` to the vector. To prevent people modifiying
```

Comment:
nit: this is not important at all, so its cool if you don't change it
```suggestion
 * need to pass a `shared_ptr` to the vector. To prevent the modification of
```